src/Illuminate/Console/Scheduling/ScheduleRunCommand.php

```diff
@@ -86,12 +77,13 @@ public function handle()
     /**
      * Run the given single server event.
      *
+     * @param  \Illuminate\Console\Scheduling\Schedule $schedule
      * @param  \Illuminate\Console\Scheduling\Event  $event
      * @return void
      */
-    protected function runSingleServerEvent($event)
+    protected function runSingleServerEvent($schedule, $event)
```

Comment:
Perhaps rather than continuously passing `$schedule` around `ScheduleRunCommand` internally, it can be set once inside the `handle()` method - i.e. `$this->schedule = $schedule`.

Then all functions within the run command have access to that variable?

Reply:
That would then leave most of this class untouched, and your just moving the Class injection and storage from the constructor to the `handle()`.

Reply:
I agree.